projects/PdfDing/services/pdfding/module.nix

```diff
@@ -0,0 +1,454 @@
+            - https://github.com/mrmn2/PdfDing/blob/d0f21ec2f9fbee4b1a2f6b7e0e6c7ea7784ab1bc/pdfding/base/task_helpers.py#L5
+            - https://huey.readthedocs.io/en/latest/api.html#crontab
+        '';
+      };
+    };
+
+    backup = {
+      enable = mkOption {
+        type = types.bool;
+        default = false;
+        description = ''
+          Automatic backup of important data to a MinIO instance.
+
+          When enabled and properly configured via environment variables,
+          important data is periodically uploaded to the specified MinIO
+          instance via cronjob.
+        '';
+      };
+      schedule = mkOption {
+        type = types.str;
+        default = "0 2 * * *";
+        description = ''
+          The cron schedule for the consume task to trigger.
+          The format is "minute hour day month day_of_week"
+          Read
+            - https://github.com/mrmn2/PdfDing/blob/d0f21ec2f9fbee4b1a2f6b7e0e6c7ea7784ab1bc/pdfding/base/task_helpers.py#L5
+            - https://huey.readthedocs.io/en/latest/api.html#crontab
+        '';
+      };
+      endpoint = mkOption {
+        type = types.nullOr types.str;
+        default = null;
+        description = "The Minio endpoint for backups";
+        example = "127.0.0.1:9000";
+      };
+    };
+
+    openFirewall = lib.mkOption {
+      type = types.bool;
+      default = false;
+      description = "Open ports in the firewall for the PdfDing web interface.";
+    };
+
+    installWrapper = mkOption {
+      type = types.bool;
+      default = false;
+      description = ''
+        This will add pdfding-manage django admin cli with proper credentials configured
+        to environment.systemPackages
+      '';
+    };
+
+    installTestHelpers = mkOption {
+      type = types.bool;
+      default = false;
+      internal = true;
+      description = "Adds a few helper commands to systemPackages for nixos tests";
+    };
+  };
+
+  config = lib.mkIf cfg.enable {
+    assertions = [
+      {
+        assertion = cfg.secretKeyFile != null;
+        message = "services.pdfding.secretKeyFile must be set when using PdfDing";
+      }
+      {
+        assertion = cfg.backup.enable -> envVars.BACKUP_ENDPOINT != null;
+        message = "services.pdfding.extraEnvironment.BACKUP_ENDPOINT must be set when backup is enabled";
+      }
+      {
+        assertion = cfg.database.createLocally -> usePostgres;
+        message = "services.pdfding.database.createLocally is enabled but not database.type is not postgres";
+      }
+      {
+        assertion = cfg.database.createLocally -> cfg.database.host == "";
+        message = "services.pdfding.database.host must be empty when services.pdfding.database.createLocally is enabled";
+      }
+      {
+        assertion = cfg.database.createLocally -> cfg.database.passwordFile == null;
+        message = "specifying services.pdfding.database.passwordFile is not supported when used along with a local db setup";
+      }
```

Comment:
I local database is used, the system (service) user must be the same as postgresql user in order get Postgresql peer authentication working.